pkg/nodediscovery/nodediscovery.go

```diff
@@ -282,7 +282,6 @@ func (n *NodeDiscovery) UpdateCiliumNodeResource() {
 
 		nodeResource.Spec.ENI.VpcID = vpcID
 		nodeResource.Spec.ENI.FirstInterfaceIndex = getInt(defaults.ENIFirstInterfaceIndex)
-		nodeResource.Spec.ENI.PreAllocate = defaults.IPAMPreAllocation
```

Comment:
I've noticed that line 222 is overwriting this variable in case we error for any possible reason. Is this intentional? More info: https://github.com/cilium/cilium/issues/9806

Reply:
I don't understand the line 222 reference but yes, #9806 is a bug that needs to be fixed separately. It's not related to this change though. This change is needed separately from the fix needed for #9806.